src/cryptonote_core/service_node_list.cpp

```diff
@@ -1035,19 +1046,33 @@ namespace service_nodes
         [&parsed_contribution](const service_node_info::contributor_t& contributor) { return contributor.address == parsed_contribution.address; });
     const bool new_contributor = (contrib_iter == contributors.end());
 
-    if (new_contributor)
+    // Check node contributor counts
```

Comment:
The `auto &info = duplicate_info(iter->second)` above is being called too early before we know we need to make a copy.

Reply:
Fixed.